Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+
+            return input;
+        }
+        protected internal virtual void HtmlAttributeEncode (string value, TextWriter output)
+        {
+
+            if (output == null)
+                throw new ArgumentNullException ("output");
+
+            if (String.IsNullOrEmpty (value))
+                return;
+
+            output.Write (HtmlAttributeEncode (value));
+        }
+
+        protected internal virtual void HtmlDecode (string value, TextWriter output)
+        {
+            if (output == null)
+                throw new ArgumentNullException ("output");
+
+            output.Write (HtmlDecode (value));
+        }
+
+        protected internal virtual void HtmlEncode (string value, TextWriter output)
+        {
+            if (output == null)
+                throw new ArgumentNullException ("output");
+
+            output.Write (HtmlEncode (value));
+        }
+
+        protected internal virtual byte[] UrlEncode (byte[] bytes, int offset, int count)
+        {
+            return UrlEncodeToBytes (bytes, offset, count);
+        }
+
+        static HttpEncoder GetCustomEncoderFromConfig ()
+        {
+#if MOBILE || NO_SYSTEM_WEB_DEPENDENCY || true
+            return defaultEncoder.Value;
+#else
+            var cfg = HttpRuntime.Section;
+            string typeName = cfg.EncoderType;
+
+            if (String.Compare (typeName, "System.Web.Util.HttpEncoder", StringComparison.OrdinalIgnoreCase) == 0)
+                return Default;
+
+            Type t = Type.GetType (typeName, false);
+            if (t == null)
+                throw new ConfigurationErrorsException (String.Format ("Could not load type '{0}'.", typeName));
+
+            if (!typeof (HttpEncoder).IsAssignableFrom (t))
+                throw new ConfigurationErrorsException (
+                    String.Format ("'{0}' is not allowed here because it does not extend class 'System.Web.Util.HttpEncoder'.", typeName)
+                );
+
+            return Activator.CreateInstance (t, false) as HttpEncoder;
+#endif
+        }
+        protected internal virtual
+        string UrlPathEncode (string value)
+        {
+            if (String.IsNullOrEmpty (value))
+                return value;
+
+            MemoryStream result = new MemoryStream ();
+            int length = value.Length;
+            for (int i = 0; i < length; i++)
+                UrlPathEncodeChar (value [i], result);
+
+            return Encoding.ASCII.GetString (result.ToArray ());
+        }
+
+        internal static byte[] UrlEncodeToBytes (byte[] bytes, int offset, int count)
+        {
+            if (bytes == null)
+                throw new ArgumentNullException ("bytes");
+
+            int blen = bytes.Length;
+            if (blen == 0)
+                return new byte [0];
+
+            if (offset < 0 || offset >= blen)
+                throw new ArgumentOutOfRangeException("offset");
+
+            if (count < 0 || count > blen - offset)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)